e2e/metricbeat_test.go

```diff
@@ -32,44 +29,62 @@ func init() {
 // the service to be monitored
 //nolint:unused
 type MetricbeatTestSuite struct {
-	cleanUpTmpFiles   bool   // if it's needed to clean up temporary files
-	configurationFile string // the  name of the configuration file to be used in this test suite
-	IndexName         string // the unique name for the index to be used in this test suite
-	ServiceName       string // the service to be monitored by metricbeat
-	ServiceType       string // the type of the service to be monitored by metricbeat
-	ServiceVariant    string // the variant of the service to be monitored by metricbeat
-	ServiceVersion    string // the version of the service to be monitored by metricbeat
-	Version           string // the metricbeat version for the test
+	cleanUpTmpFiles   bool               // if it's needed to clean up temporary files
+	configurationFile string             // the  name of the configuration file to be used in this test suite
+	ServiceName       string             // the service to be monitored by metricbeat
+	ServiceType       string             // the type of the service to be monitored by metricbeat
+	ServiceVariant    string             // the variant of the service to be monitored by metricbeat
+	ServiceVersion    string             // the version of the service to be monitored by metricbeat
+	Query             ElasticsearchQuery // the specs for the ES query
+	Version           string             // the metricbeat version for the test
+}
+
+// getIndexName returns the index to be used when querying Elasticsearch
+func (mts *MetricbeatTestSuite) getIndexName() string {
+	return mts.Query.IndexName
+}
+
+func (mts *MetricbeatTestSuite) setEventModule(eventModule string) {
+	mts.Query.EventModule = eventModule
 }
 
 // As we are using an index per scenario outline, with an index name formed by metricbeat-version1-module-version2,
+// or metricbeat-version1-module-variant-version2,
```

Comment:
Ok to continue with variants as something special by now, but take into account that there may be more variables, for example we already have a "codename" in ceph tests.

Reply:
Let's find in a follow-up PR a better way of index naming based on the discovered variables.